getinsomnia/insomnia
# feat: enable insomnia.sendRequest in pre-request scripting - INS-3379

### Please review this PR (https://github.com/Kong/insomnia/pull/7145) firstly as current one is based on it

Changes:
- [x] enabled `insomnia.sendRequest`


### How to test
Run this script and check if it runs well, basically it sends several requests with different content types:
```
            const rawReq = {
                url: 'https://httpbin.org/anything',
                method: 'POST',
                header: {
                    'Content-Type': 'text/plain',
                },
                body: {
                    mode: 'raw',
                    raw: 'rawContent',
                },
            };
            const urlencodedReq = {
                url: 'https://httpbin.org/anything',
                method: 'POST',
                header: {
                    'Content-Type': 'application/x-www-form-urlencoded',
                },
                body: {
                    mode: 'urlencoded',
                    urlencoded: [
                        { key: 'k1', value: 'v1' },
                        { key: 'k2', value: 'v2' },
                    ],
                },
            };
            const gqlReq = {
                url: 'https://httpbin.org/anything',
                method: 'POST',
                header: {
                    'Content-Type': 'application/graphql',
                },
                body: {
                    mode: 'graphql',
                    graphql: {
                        query: 'query',
                        operationName: 'operation',
                        variables: 'var',
                    },
                },
            };
            const fileReq = {
                url: 'https://httpbin.org/anything',
                method: 'POST',
                header: {
                    'Content-Type': 'application/octet-stream',
                },
                body: {
                    mode: 'file',
                    file: "${getFixturePath('files/rawfile.txt')}",
                },
            };
            const formdataReq = {
                url: 'https://httpbin.org/anything',
                method: 'POST',
                header: {
                    // 'Content-Type': 'multipart/form-data',
                },
                body: {
                    mode: 'formdata',
                    formdata: [
                        { key: 'k1', type: 'text', value: 'v1' },
                        { key: 'k2', type: 'file', value: "${getFixturePath('files/rawfile.txt')}" },
                    ],
                },
            };
            const promises = [rawReq, urlencodedReq, gqlReq, fileReq, formdataReq].map(req => {
                return new Promise((resolve, reject) => {
                    insomnia.sendRequest(
                        req,
                        (err, resp) => {
                            if (err != null) {
                                reject(err);
                            } else {
                                resolve(resp);
                            }
                        }
                    );
                });
            });
            // send request
            const resps = await Promise.all(promises);
            // set envs
            insomnia.environment.set('rawBody', resps[0].body);
            insomnia.environment.set('urlencodedBody', resps[1].body);
            insomnia.environment.set('gqlBody', resps[2].body);
            insomnia.environment.set('fileBody', resps[3].body);
            insomnia.environment.set('formdataBody', resps[4].body);
```
